xmtp_mls/src/storage/sql_key_store.rs

```diff
@@ -22,60 +31,951 @@ impl<'a> SqlKeyStore<'a> {
     pub fn conn(&self) -> &DbConnection<'a> {
         self.conn
     }
-}
 
-impl OpenMlsKeyStore for SqlKeyStore<'_> {
-    /// The error type returned by the [`OpenMlsKeyStore`].
-    type Error = StorageError;
-
-    /// Store a value `v` that implements the [`MlsEntity`] trait for
-    /// serialization for ID `k`.
-    ///
-    /// Returns an error if storing fails.
-    fn store<V: MlsEntity>(&self, k: &[u8], v: &V) -> Result<(), Self::Error> {
-        self.conn()
-            .insert_or_update_key_store_entry(k.to_vec(), db_serialize(v)?)?;
+    pub fn write<const VERSION: u16>(
```

Comment:
Right, so the easiest transition is to keep the generic read/write/delete functions in some way and use them in all the other functions.
I'm sure you'll want to do something more intelligent later though.